File: modules/spring-web/testdata/java/loader/src/ProductFeignClient.java
package com

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

@FeignClient(value = "productFeignClient")
public interface ProductFeignClient {
    @PostMapping(value = "product")
    Object product(@RequestBody Object dto);
}

File: modules/spring-web/testdata/kotlin/loader/src/ProductFeignClient.kt
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

@FeignClient(value = "productFeignClient")
interface ProductFeignClient {
    @PostMapping(value = "product")
    fun product(@RequestBody dto: Object): Object
}